### PR TITLE
Update tree-sitter and each grammars to v0.19

### PR DESCRIPTION
**Pull request by Luni-4**
This PR updates the `tree-sitter` crate and each `rust-code-analysis` grammar to `v0.19`

Ticks an item when there are no significant differences in the grammar
- [x] tree-sitter-mozcpp
- [x] tree-sitter-mozjs
- [x] tree-sitter-ccomment
- [x] tree-sitter-preproc
- [x]  tree-sitter-rust
- [x] tree-sitter-python
- [x] tree-sitter-javascript
- [x] tree-sitter-typescript

### CODE REVIEW

File: src/metrics/loc.rs [hunk 1]
```diff
@@ -755,4 +755,25 @@ mod tests {
...
+                 return;
+               }
+             });",
+            "foo.js",
+            JavascriptParser,
+            loc,
+            [
+                (sloc, 5, usize),  // The number of lines is 5
+                (ploc, 5, usize),  // The number of code lines is 5
+                (lloc, 7, usize),  // The number of statements is 7
```
Comment (Luni-4) [inline src/metrics/loc.rs:773]: @marco-c I don't if I counted correctly the number of statements. Could you re-check, please?

**Comment by Luni-4** [comment]
@marco-c 

This PR is ready to be reviewed. I covered all possible cases extracted from minimal tests and notified upstream when there were parse errors.

**Comment by marco-c** [comment]
> This PR is ready to be reviewed. I covered all possible cases extracted from minimal tests and notified upstream when there were parse errors.

Can you list here in a comment all the issues you filed upstream? So we have a way to track them.

**Comment by Luni-4** [comment]
List of regressions that have been found for `tree-sitter v0.19`

- Rust: https://github.com/tree-sitter/tree-sitter-rust/issues/106
- JavaScript: https://github.com/tree-sitter/tree-sitter-javascript/issues/164 https://github.com/tree-sitter/tree-sitter-javascript/issues/165
- TypeScript: https://github.com/tree-sitter/tree-sitter-typescript/issues/139
- Mozcpp: #579

File: src/checker.rs [hunk 2]
```diff
@@ -212,7 +212,17 @@ impl Checker for TypescriptCode {
         ClassDeclaration,
         ArrowFunction
     );
-    mk_else_if!(IfStatement);
```
Comment (marco-c) [inline src/checker.rs:215]: It seems for both Rust and TypeScript you made the same change. What if we changed the `mk_else_if!` macro instead? Is there any language which still needs the previous implementation?

**Review by marco-c — APPROVED** [review]

File: src/checker.rs (hunk 2 above)
Comment (Luni-4) [reply src/checker.rs:215]: JavaScript, Python,  Mozcpp needs the previous implementation

**Pull request merged**